### PR TITLE
Merging ZIdebug into Develop to fix issue #139 of QuSurf Issue tracker

### LINKED ISSUE #139: Include "robust" ramsey find frequency as an option in find_frequency
We ( @elrama-  and me)  have found the following snippet very useful. It should be integrated in the qubit object. 
```python 
    times = np.arange(0, .1e-6, 1e-9)
    artificial_detuning = 2/times[-1]

    for i in range(2):
        QR.measure_ramsey(times, artificial_detuning=artificial_detuning)
        a = ma.Ramsey_Analysis(auto=True, close_fig=True)
        fitted_freq = a.fit_res.params['frequency'].value
        detuning = fitted_freq-artificial_detuning
        QR.f_qubit(QR.f_qubit()-detuning)
```

### PR DESCRIPTION
**Pull request by lmjanssen**
Please use the following template for a pull request. 

Fixes #139 of QuSurf Issue tracker.
Also includes the option to use Use_min in the nested resonator scan of a qubit flux scan

Changes proposed in this pull request:
- Timing issue with UHFQC is resolved (See issue #139 in QuSurf Issue tracker)
- Added a Use_min option to the nested resonator scans for a DAC scan

The changes below are already in develop:
- Fix label issue for spectroscopy
- Added target value for the offset calibration

@AdriaanRol could you check this please?

In order for the pull request to be merged, the following conditions must be met:
- travis test suite passes
- all reasonable issues raised by codacy must be resolved 
- a positive review is required

Whenever possible the pull request should
- follow the PEP8 style guide 
- have tests for the code
- be well documented and contain comments

Tests are not mandatory as this is generally hard to make for instruments that interact with hardware.

### CODE REVIEW

File: pycqed/analysis/measurement_analysis.py [hunk 1]
```diff
@@ -7098,12 +7098,15 @@ def run_default_analysis(self, print_fit_results=False, analyze_ef=False,
             except (TypeError, KeyError, ValueError):
                 logging.warning('qb_name is None. Old parameter values will '
                                 'not be retrieved.')
-                label = 'f0={:.5f} GHz $\pm$ ({:.2f}) MHz ' \
-                        '\nkappa0={:.4f} MHz $\pm$ ({:.2f}) MHz'.format(
-                            self.fit_res.params['f0'].value * scale,
-                            self.fit_res.params['f0'].stderr / 1e6,
-                            self.fit_res.params['kappa'].value / 1e6,
-                            self.fit_res.params['kappa'].stderr / 1e6)
+                try: #Dirty fix, should already be fine in Develop
```
Comment (AdriaanRol) [inline pycqed/analysis/measurement_analysis.py]: This is not how it should be fixed and this should not be part of this PR. 

Could you add the proper fix after merging? The proper fix uses the function `format_value_string` that can be found in `pycqed/utilities/general` examples of its use can be found in for example `analysis_v2/timedomain_analysis`

File: pycqed/instrument_drivers/meta_instrument/qubit_objects/CCL_Transmon.py [hunk 2]
```diff
@@ -1803,6 +1803,7 @@ def measure_qubit_frequency_dac_scan(self, freqs, dac_values,
                                          pulsed=True, MC=None,
                                          analyze=True, fluxChan=None, close_fig=True,
                                          nested_resonator_calibration=False,
+                                         nested_resonator_calibration_use_min=False,
```
Comment (AdriaanRol) [inline pycqed/instrument_drivers/meta_instrument/qubit_objects/CCL_Transmon.py]: Change does not belong in this PR but I'm fine with merging.

File: pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_HDAWG8.py [hunk 3]
```diff
@@ -447,19 +481,35 @@ def upload_waveform_realtime(self, w0, w1, awg_nr: int, wf_nr: int =1):
         self._realtime_w0 = w0
         self._realtime_w1 = w1
 
+        print('wf_nr', wf_nr)
```
Comment (AdriaanRol) [inline pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_HDAWG8.py]: This print statement should be removed. Otherwise we completely spam the terminal in daily use.

File: pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_HDAWG8.py [hunk 4]
```diff
@@ -447,19 +481,35 @@ def upload_waveform_realtime(self, w0, w1, awg_nr: int, wf_nr: int =1):
...
+        print('w0', len(w0))
+        print('w1', len(w1))
+
         # Checked and everything matches
```
Comment (AdriaanRol) [inline pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_HDAWG8.py]: Commented out code like this should be removed.

File: pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_HDAWG8.py [hunk 5]
```diff
@@ -447,19 +481,35 @@ def upload_waveform_realtime(self, w0, w1, awg_nr: int, wf_nr: int =1):
...
         #print(hex(id(self._dev)))
 
-        c = np.vstack((w0, w1)).reshape((-2,), order='F')
-        self._dev.seti('awgs/{}/enable'.format(awg_nr), 0)
-        self._dev.subs('awgs/{}/ready'.format(awg_nr))
-        self._dev.seti('awgs/{}/waveform/index'.format(awg_nr), wf_nr)
-        #self._dev.setv('awgs/{}/waveform/data'.format(awg_nr), c)
-        # Try as float32 instead
-        self._dev.setv('awgs/{}/waveform/data'.format(awg_nr), c.astype(np.float32))
+        # Changed by NielsH (niels@zhinst.com)
```
Comment (AdriaanRol) [inline pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_HDAWG8.py]: Commented out code should be removed.

File: pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_HDAWG8.py [hunk 6]
```diff
@@ -620,13 +670,20 @@ def configure_codeword_protocol(self, default_dio_timing: bool=False):
             # Valid polarity is 'high' (hardware value 2),
             # 'low' (hardware value 1), 'no valid needed' (hardware value 0)
             self.set('awgs_{}_dio_valid_polarity'.format(awg_nr), 2)
+            
+            # NH: With the new DIO protocol the strobe is no longer directly used!
```
Comment (AdriaanRol) [inline pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_HDAWG8.py]: Then let's remove these 3 lines.

File: pycqed/measurement/calibration_toolbox.py [hunk 7]
```diff
@@ -75,6 +75,7 @@ def mixer_carrier_cancellation(SH, source, MC,
                     'sigma0':1,
                     'options': {'maxiter': 300,    # maximum function cals
                                 # Scaling for individual sigma's
+                                'ftarget': -110,
```
Comment (AdriaanRol) [inline pycqed/measurement/calibration_toolbox.py]: Does not belong in this PR but a good change anyway. Maybe add a comment why this is the right target for optimization.

File: pycqed/analysis/measurement_analysis.py (hunk 1 above)
Comment (lmjanssen) [reply pycqed/analysis/measurement_analysis.py]: Will fix after merging

File: pycqed/instrument_drivers/meta_instrument/qubit_objects/CCL_Transmon.py (hunk 2 above)
Comment (lmjanssen) [reply pycqed/instrument_drivers/meta_instrument/qubit_objects/CCL_Transmon.py]: Agreed

File: pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_HDAWG8.py (hunk 4 above)
Comment (lmjanssen) [reply pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_HDAWG8.py]: Done

File: pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_HDAWG8.py (hunk 5 above)
Comment (lmjanssen) [reply pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_HDAWG8.py]: Done

File: pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_HDAWG8.py (hunk 6 above)
Comment (lmjanssen) [reply pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_HDAWG8.py]: Done

File: pycqed/measurement/calibration_toolbox.py (hunk 7 above)
Comment (lmjanssen) [reply pycqed/measurement/calibration_toolbox.py]: Comment added: -110 is the noise level of the SH

Comment (lmjanssen) [reply pycqed/measurement/calibration_toolbox.py]: Conflict is in this file: will delete from this commit, and update in Develop after merging
